File: Ruby/Shield.rb
#encoding: UTF-8

require_relative 'Dice'
require_relative 'CombatElement'

module Irrgarten
	class Shield < CombatElement
		def initialize(protection,uses)
			super(protection,uses)
		end	
		
		public
		def protect()
			produce_effect()
		end
		
		def to_string
			"S"+super
		end
		
		public_class_method :new
	end
end

File: Java/src/irrgarten/Shield.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 * @author jorge
 */

public class Shield extends CombatElement{
    // Constructor
    public Shield (float protection, int uses){
        super(protection,uses);
    }
    
    
    // Métodos públicos
    public float protect (){
        return produceEffect();
    }
    
    @Override
    public String toString(){
        return "S"+super.toString();
    }
}